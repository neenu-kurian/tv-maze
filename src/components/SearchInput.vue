<template>
  <div
    role="search"
    class="flex align-center gap-6 rounded-md border-transparent px-4 py-2 md:w-[60%] max-w-full bg-gray-100 width-full"
  >
    <slot name="icon">
      <MagnifyingGlassIcon class="w-8 h-8 flex-none basis-5" />
    </slot>
    <input
      :value="modelValue"
      name="search-input"
      type="search"
      autocomplete="off"
      aria-label="Search shows"
      @input="emit('update:modelValue', ($event.target as HTMLInputElement).value)"
      @keydown.enter="emit('enter', $event)"
      :placeholder="placeholder"
      class="border-none outline-none bg-transparent w-full text-base text-gray-900 p-0 leading-none"
    />
  </div>
</template>

<script setup lang="ts">
import { MagnifyingGlassIcon } from '@heroicons/vue/24/outline';
import type { SearchInputProps } from '@/types/index.ts';

withDefaults(defineProps<SearchInputProps>(), {
  modelValue: '',
  placeholder: 'Search for TV shows...',
});

const emit = defineEmits<{
  'update:modelValue': [value: string];
  enter: [event: Event];
}>();
</script>
