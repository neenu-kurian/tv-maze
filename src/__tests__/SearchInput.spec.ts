import { render, screen } from '@testing-library/vue';
import SearchInput from '@/components/SearchInput.vue';
import { describe, it, expect, vi } from 'vitest';
import '@testing-library/jest-dom';

vi.mock('@heroicons/vue/24/outline', () => ({
  MagnifyingGlassIcon: {
    template: '<div data-testid="search-icon"></div>',
  },
}));

const renderComponent = () =>
  render(SearchInput, {
    props: {
      modelValue: '',
      placeholder: 'Search for TV shows...',
    },
  });

describe('SearchInput.vue', () => {
  it('matches snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('renders the search input with default props', () => {
    renderComponent();
    const input = screen.getByRole('searchbox', { name: 'Search shows' });
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'search');
    expect(input).toHaveAttribute('placeholder', 'Search for TV shows...');
    expect(input).toHaveValue('');
  });
});
