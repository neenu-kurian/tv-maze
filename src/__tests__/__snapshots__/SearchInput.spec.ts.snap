// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex align-center gap-6 rounded-md border-transparent px-4 py-2 md:w-[60%] max-w-full bg-gray-100 width-full"
    role="search"
  >
    
    <div
      class="w-8 h-8 flex-none basis-5"
      data-testid="search-icon"
    />
    
    <input
      aria-label="Search shows"
      autocomplete="off"
      class="border-none outline-none bg-transparent w-full text-base text-gray-900 p-0 leading-none"
      name="search-input"
      placeholder="Search for TV shows..."
      type="search"
      value=""
    />
  </div>
</div>
`;
